sdk/core/azure-core/src/main/java/com/azure/core/http/RequestConditions.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http;
+
+import java.time.OffsetDateTime;
+
+/**
+ * Specifies HTTP options for conditional requests based on modification time.
+ */
+public class RequestConditions extends MatchConditions {
+    private OffsetDateTime ifModifiedSince;
+    private OffsetDateTime ifUnmodifiedSince;
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return The updated ResourceConditions object.
+     */
+    @Override
+    public RequestConditions setIfMatch(String ifMatch) {
+        super.setIfMatch(ifMatch);
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return The updated ResourceConditions object.
```

Comment:
{@inheritDoc} + new doc doesn't generate the Javadocs correctly. Use `@inheritDoc` only or add new javadoc.

Reply:
Pulled the super class documentation class directly.